### DISCUSSION TITLE
Release channels naming convention

### DISCUSSION THREAD

**Discussion by ghost** [microsoft/winget-pkgs#14128] [Q&A] [answered]
So, I just have started committing to the package list, and one thing bothers me slightly: release channels and package IDs.

Most packages that I have seen append channel name to the package name itself, thus having separate folders for different channels. However, _PowerShell Preview_ is named `PowerShell-Preview`, with dash `-`, though it also has different folder; _Visual Studio Code Insiders_, on the other hand, is named `VisualStudioCode.Insiders` and is located in `Insiders` folder inside the main package folder.

So, I have two questions right now:

1. How should I name preview packages that I am going to add to the package list? I personally prefer the way _Visual Studio Code_ is done.
2. Should we standardize this point somehow and introduce some conventions?

I have also seen in microsoft/winget-cli#147 that _VS Code Insiders_, apparently, previously had a separate folder, but at some point between 09/30/2020 and 03/30/2021 (could not find the exact PR) its ID was modified and it was moved inside the GA folder.

So, yeah, how should I deal with release channels?

**Comment by denelon** [comment, marked as answer]
We haven't really implemented channels the way we were originally thinking yet. The current thinking with packages like Visual Studio Code is that they are a separate package that can safely live "side by side". 

With channels the idea is that it would be for packages that cannot live "side by side" like Docker Desktop. We don't want to incorrectly upgrade a stable package on top of another channel or vice versa. That implies the notion of channel being "inside" the same manifest for those packages.
